### ISSUE TITLE
docker-compose up error

### ISSUE THREAD

**Issue by nickshanks347** [boringproxy/boringproxy#80]
Linked to #77 but decided to create a new ticket because that was resolved, and subsequently closed.

Thanks for the help @fbartels and @anderspitman but now I get a different error. It seems to successfully build but on `docker-compose up`:
```
docker-compose up
Creating network "boringproxy_default" with the default driver
Creating boringproxy_boringproxy_1 ... done
Attaching to boringproxy_boringproxy_1
boringproxy_1  | standard_init_linux.go:211: exec user process caused "exec format error"
boringproxy_boringproxy_1 exited with code 1
```

After some more research, seems like an architecture issue. I'm running Ubuntu Server 64-bit. Also tried on a Pi though, didn't work either.

**Comment by fbartels** [comment]
Hi @nickshanks347,

the error message indeed suggests an architecture issue, but that should only happen if you run the image on another architecture than it was built for (which cannot really be the case if you build and run on the same machine). Additionally for me on my amd64 desktop there is no problem.

**Comment by nickshanks347** [comment]
Yeah, cloning from source and then just running docker-compose, weird stuff because it's bUilt and run on the same machine. What OS do you use because I'll make a VM on my PC and try it out.

**Comment by fbartels** [comment]
Ubuntu 20.04

**Comment by nickshanks347** [comment]
Hmm, that's what I'm using too. Know of a way to specify a build architecture in the docker file?

**Comment by nickshanks347** [comment]
Interestingly, this commit `https://github.com/boringproxy/boringproxy/tree/ada81bae62ca18f8b3e004a97c03d0f1d6d24431` seems to work, what has changed since then which stops it building the correct architecture?

**Comment by nickshanks347** [comment]
I've done a compare myself and I can't see anything which stands out, apart from some changes to the scripts/ directory but I don't think these are used anyway. I've always tried a ubuntu and Debian builder, neither worked same issue. I've also tried adding GOOS and GOARCH to the go build command. Nothing.

**Comment by fbartels** [comment]
> Know of a way to specify a build architecture in the docker file?

That is afaik not possible in the dockerfile itself, but can be passed along with the docker build command (but also not with `docker-compose`). The command could look like this: `docker buildx build --platform amd64 .`

My own experiments with boringproxy did not progress any further after my pr with the Dockerfile. So if @anderspitman does not use it, or there are no other users interested in picking it up, its probably best to remove this code again.

**Comment by nickshanks347** [comment]
I'll try and build with the docker command, and I'd pick it up but I need to learn Docker first :)

**Comment by anderspitman** [comment]
Yeah I think usage of boringproxy is still low and nobody really using Docker. I don't think it's really ready for a proper Docker deployment, because people expect Docker images to just work, and boringproxy isn't at that level regardless of how it's packaged.

**Comment by nickshanks347** [comment]
Yeah, I get what you're saying. Boringproxy is sounds pretty amazing, and I am yet to use it but I haven't found anything that does what boringproxy does. I managed to build the release then stick that in a docker container.

**Comment by nickshanks347** [comment]
Where in the src code can you change which ports it listens on?

**Comment by nickshanks347** [comment]
@anderspitman can you think of any solution which runs in docker that is _like_ boringproxy? I have looked but boringproxy seems by far the best

**Comment by anderspitman** [comment]
You've checked my list here, right? https://github.com/anderspitman/awesome-tunneling

Some of those probably provide docker. If you find any let me know and I'll add it to the notes so they're easier to find for others.

Have you tried deploying boringproxy *without* docker? It's pretty dang easy. I don't like making people depend on docker so I tried to keep it as simple as possible.

**Comment by nickshanks347** [comment]
Yeah, I've seen your list but I'll have a proper read of it later. I have tried deploying without Docker but the machine I want it to live on already has a webmail service listening on 80 and 443. I can either use docker on that server or find a dirt cheap VPS to host boringproxy on. With that being said, I wrote reddit post [here](https://www.reddit.com/r/docker/comments/mh35w3/architecture_issue_with_running_a_docker_project/) and someone got the Dockerfile to work. I'll make a fork and send a PR.
